server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CCSUsageTelemetryIT.java

```diff
@@ -0,0 +1,509 @@
+        // We want to send search to a specific node (we don't care which one) so that we could
+        // collect the CCS telemetry from it later
+        String nodeName = cluster(LOCAL_CLUSTER).getRandomNodeName();
+        // We don't care here too much about the response, we just want to trigger the telemetry collection.
+        // So we check it's not null and leave the rest to other tests.
+        assertResponse(cluster(LOCAL_CLUSTER).client(nodeName).search(searchRequest), Assert::assertNotNull);
+        return getTelemetrySnapshot(nodeName);
+    }
+
+    /**
+     * Create search request for indices and get telemetry from it
+     */
+    private CCSTelemetrySnapshot getTelemetryFromSearch(String... indices) throws ExecutionException, InterruptedException {
+        return getTelemetryFromSearch(makeSearchRequest(indices));
+    }
+
+    /**
+     * Search on all remotes
+     */
+    public void testAllRemotesSearch() throws ExecutionException, InterruptedException {
+        Map<String, Object> testClusterInfo = setupClusters();
+        String localIndex = (String) testClusterInfo.get("local.index");
+        String remoteIndex = (String) testClusterInfo.get("remote.index");
+
+        SearchRequest searchRequest = makeSearchRequest(localIndex, "*:" + remoteIndex);
+        boolean minimizeRoundtrips = TransportSearchAction.shouldMinimizeRoundtrips(searchRequest);
+
+        String nodeName = cluster(LOCAL_CLUSTER).getRandomNodeName();
+        assertResponse(
+            cluster(LOCAL_CLUSTER).client(nodeName)
+                .filterWithHeader(Map.of(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER, "kibana"))
+                .search(searchRequest),
+            Assert::assertNotNull
+        );
+        CCSTelemetrySnapshot telemetry = getTelemetrySnapshot(nodeName);
+
+        assertThat(telemetry.getTotalCount(), equalTo(1L));
+        assertThat(telemetry.getSuccessCount(), equalTo(1L));
+        assertThat(telemetry.getFailureReasons().size(), equalTo(0));
+        assertThat(telemetry.getTook().count(), equalTo(1L));
+        assertThat(telemetry.getTookMrtTrue().count(), equalTo(minimizeRoundtrips ? 1L : 0L));
+        assertThat(telemetry.getTookMrtFalse().count(), equalTo(minimizeRoundtrips ? 0L : 1L));
+        assertThat(telemetry.getRemotesPerSearchAvg(), equalTo(2.0));
+        assertThat(telemetry.getRemotesPerSearchMax(), equalTo(2L));
+        assertThat(telemetry.getSearchCountWithSkippedRemotes(), equalTo(0L));
+        assertThat(telemetry.getClientCounts().size(), equalTo(1));
+        assertThat(telemetry.getClientCounts().get("kibana"), equalTo(1L));
+        if (minimizeRoundtrips) {
+            assertThat(telemetry.getFeatureCounts().get(MRT_FEATURE), equalTo(1L));
+        } else {
+            assertThat(telemetry.getFeatureCounts().get(MRT_FEATURE), equalTo(null));
+        }
```

Comment:
Can you also add an assert here that the ASYNC feature count is not incremented (since you have tests for that feature in the Async IT test).